### PR TITLE
feat: Add option to filter alerts by labels

### PR DESCRIPTION
**Pull request by mimotej**
This PR introduces a new annotation called `prometheus.io/labels`. This annotation is optional, and when used, it filters alerts by labels that are present. It filters an array of alerts created using a `prometheus.io/alert` annotation. Labels are defined in the following format:
```
prometheus.io/labels: "label1=value1,label2=value2"
```
Alerts have to have all the labels defined, or they are not included in the final array.

#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [x] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

File: .changeset/light-melons-share.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+'@roadiehq/backstage-plugin-prometheus': major
```
Comment (Xantier) [inline .changeset/light-melons-share.md:2]: This doesn't look like a breaking change so a `minor` bump is enough when introducing new functionality I think

**Review by Xantier — DISMISSED** [review]
Excellent. Thanks for the contribution. I'll merge this in after CI and after the change to a minor release

File: .changeset/light-melons-share.md (hunk 1 above)
Comment (mimotej) [reply .changeset/light-melons-share.md]: Good point :+1:. Fixed thank you

**Review by Xantier — APPROVED** [review]
:shipit:

**Pull request merged**